front/src/main.js

```diff
@@ -0,0 +1,423 @@
+
+Vue.use(Vuex)
+Vue.use(VueRouter)
+
+Vue.use(VueTimeago, {
+  name: 'timeago',
+  locale: 'en-US',
+  locales: {
+    'en-US': require('vue-timeago/locales/en-US.json')
+  },
+})
+
+const routes = [
+  {
+    path: '/',
+    redirect: '/dashboard',
+  },
+  {
+    path: '/dashboard',
+    component: Dashboard,
+  },
+  {
+    path: '/games',
+    component: GameList,
+  },
+  {
+    path: '/teams',
+    component: TeamList,
+  },
+  {
+    path: '/matches',
+    component: MatchList,
+  },
+  {
+    path: '/tournaments',
+    component: TournamentList,
+  },
+  {
+    path: '/login',
+    component: Login,
+  },
+  {
+    path: '/logout',
+    redirect: '/login',
+  },
+  {
+    path: '/tournaments/:id',
+    name: 'tournament',
+    component: Tournament,
+  },
+  {
+    path: '/matches/:id',
+    name: 'match',
+    component: Match,
+  },
+  {
+    path: '/teams/:id',
+    name: 'team',
+    component: Team,
+  },
+]
+
+export var router = new VueRouter({
+  routes
+})
+
+function enrichMatch(match, getters) {
+  return Vue.util.extend(match, {
+    teamOne: getters.teamMap.get(match.team_one_id),
+    teamTwo: getters.teamMap.get(match.team_two_id),
+    game: getters.gameMap.get(match.game_id),
+  })
+}
+
+function enrichFriendlyMatch(match, getters) {
+  match = enrichMatch(match, getters)
+  match.owner = getters.userMap.get(match.owner_id)
+  return match
+}
+
+export var store = new Vuex.Store({
+  state: {
+    currentUser: null,
+    games: [],
+    teams: [],
+    matches: [],
+    tournaments: [],
+    users: [],
+  },
+  getters: {
+    isLoggedIn (state) {
+      return state.currentUser !== null
+    },
+    isAdmin (state) {
+      return state.currentUser && state.currentUser.role === 'admin'
+    },
+    currentUser (state) {
+      return state.currentUser
+    },
+
+    userList (state) {
+      return state.users
+    },
+    userMap (state, getters) {
+      return new Map(getters.userList.map(user => [user.id, user]))
+    },
+
+    // rawGameList (state) {
+    // },
+    gameList (state, getters) {
+      return state.games
+    },
+    gameMap (state, getters) {
+      return new Map(getters.gameList.map(game => [game.id, game]))
+    },
+
+    rawTeamList (state) {
+      return state.teams
+    },
+    teamList (state, getters) {
+      return getters.rawTeamList.map(team => {
+        return Vue.util.extend(team, {
+          members: team.member_ids.map(memberId => getters.userMap.get(memberId))
+        })
+      })
+    },
+    teamMap (state, getters) {
+      return new Map(getters.teamList.map(team => [team.id, team]))
+    },
+
+    rawMatchList (state) {
+      return state.matches
+    },
+    matchList (state, getters) {
+      return getters.rawMatchList.map(match => enrichFriendlyMatch(match, getters))
+    },
+    matchMap (state, getters) {
+      return new Map(getters.matchList.map(match => [match.id, match]))
+    },
+
+    // need or not?
+    userMatches (state) {
+      if(state.currentUser) {
```

Comment:
It is a common practice to do early returns to reduce unnecessary indentation. In this case:
```javascript
if (!state.currentUser) return []

let userId = state.currentUser.id
// rest of te code
```